### PR TITLE
Rust connector test kit

### PR DESCRIPTION
**Pull request by dpetrick**
See [Readme](https://github.com/prisma/prisma-engines/blob/2c1435aa38bbf102e20d7c02ac184cb2273b253e/query-engine/connector-test-kit-rs/README.md) for details.

### CODE REVIEW

File: query-engine/connector-test-kit-rs/query-engine-tests/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+// The mod name dictates the db name. If the name is `some_spec`
+// then, for example, the MySQL db should be (similar to) `some_spec` as well.
+// #[cfg(test)]
+// #[before_each(before_each_handler)] // Hook to run before each test.
+// #[schema(schema_handler)] // Schema for all contained tests. Allows us to cache runners maybe.
+#[test_suite(schema(schemas::some_common_schema), only(Postgres(9)))]
+mod some_spec {
+
+    // These imports are required if no `#[test_suite(...)]` is used
```
Comment (pimeys) [inline query-engine/connector-test-kit-rs/query-engine-tests/src/lib.rs:32]: You should add the imports to the macro instead. Better to have them there than in ever test file...

File: .buildkite/engineer [hunk 2]
```diff
@@ -13,7 +13,7 @@ fi
 if ! type "engineer" > /dev/null; then
     # Setup Prisma engine build & test tool (engineer).
     set -e
-    curl --fail -sSL "https://prisma-engineer.s3-eu-west-1.amazonaws.com/1.14/latest/$OS/engineer.gz" --output engineer.gz
```
Comment (dpetrick) [inline .buildkite/engineer:16]: Todo: When ready for merge, engineer will be bumped accordingly.

File: query-engine/connector-test-kit-rs/query-test-macros/src/connector_test.rs [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+    test.into()
+}
+
+fn quote_connector(tag: ConnectorTag) -> proc_macro2::TokenStream {
+    let (connector, version) = tag.as_parse_pair();
+
+    match version {
+        Some(version) => quote! {
+            ConnectorTag::try_from((#connector, Some(#version))).unwrap()
```
Comment (dpetrick) [inline query-engine/connector-test-kit-rs/query-test-macros/src/connector_test.rs:110]: This is fairly idiotic and build at the very start of my macro adventure. Can be redone later.

File: query-engine/connector-test-kit-rs/query-tests-setup/src/connector_tag/mod.rs [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    ///   implementing connector, like a file or a schema.
+    /// - `is_ci` signals whether or not the test run is done on CI or not. May be important if local
+    ///   test run connection strings and CI connection strings differ because of networking.
+    fn connection_string(&self, test_database: &str, is_ci: bool) -> String;
+
+    /// Capabilities of the implementing connector.
+    fn capabilities(&self) -> &[ConnectorCapability];
+
+    /// Serialization of the connector. Expected to return `(tag_name, version)`.
+    /// Todo: Think of something better.
```
Comment (dpetrick) [inline query-engine/connector-test-kit-rs/query-tests-setup/src/connector_tag/mod.rs:38]: Ties into the comment above how this should not be a thing.

**Pull request merged**